lib/deprecation.nix

```diff
@@ -122,4 +122,12 @@ rec {
       nestedTypes.coercedType = oldType;
       nestedTypes.finalType = newType;
     };
+
+  mkRemovedPackageOptionModule =
+    pluginName: packageName:
+    lib.mkRemovedOptionModule [ "plugins" pluginName "${packageName}Package" ] ''
+      Please use the `dependencies.${packageName}` top-level option instead:
+      - `dependencies.${packageName}.enable = false` to disable installing `${packageName}`
+      - `dependencies.${packageName}.package` to choose which package to install for `${packageName}`.
+    '';
```

Comment:
It would not be an excessive amount of work to write our own module here that offers a more graceful migration period.

Such a change likely wouldn't affect the treewide diff.

Is this something you're opposed to in principal, or something you aren't interested in working on personally? I won't block the PR on this point, but I might be interested in tackling it as follow-up or parallel work.